fcrepo-exts/fcrepo-camel-toolbox
# FCREPO-3770: Allow to configure basic authentication for a triple store

# What does this Pull Request do?
With this PR, the toolbox will be able to talk with triple stores that require basic authentication.

# What's new?

* Updated the Jena libraries to v3.17.0 so the on-demand Fuseki server we use for testing can do basic authentication.
* Depending if `triplestore.authUsername` is set, the Triplestore Router will set a basic authentication header.
* Added integration tests for both `update` and `delete` with authentication.

# How should this be tested?

Setup a Fedora as per README:

```
docker run -p8080:8080 --rm -p61616:61616  -p8181:8181 --name=my_fcrepo6  fcrepo/fcrepo:6.0.0
```

Run Zazuko's Fuseki:

```
docker run --name fuseki -p3030:3030 --rm -e ADMIN_PASSWORD=fuseki -e OTEL_JAVAAGENT_ENABLED=false zazuko/jena-fuseki:4.1.0 --mem /test
```

Create a properties file with the following properties:

```
triplestore.indexer.enabled=true
solr.indexer.enabled=true
triplestore.baseUrl=http://localhost:3030/test
triplestore.authUsername=admin
triplestore.authPassword=fuseki
solr.baseUrl=http://localhost:8983/solr/
```

Build the project (note that the tests currently don't work, so `-DskipTests -DskipITs` should be set) and run it:

```
mvn clean install
java -jar fcrepo-camel-toolbox-app/target/fcrepo-camel-toolbox-app-6.0.0-SNAPSHOT-driver.jar -c ./application.properties
```

Post some data to Fedora:

```
curl -X POST -u fedoraAdmin:fedoraAdmin -d "$(cat fcrepo-indexing-triplestore/src/test/resources/indexable.ttl)" http://localhost:8080/fcrepo/rest
```

Afterwards, check the logs of all components. The toolbox should log a successful request to Fedora and Fuseki. You can also do a SparQL query to verify that the data has been inserted.

# Additional Notes:
I tried to copy the integration test which tests the update, but for whatever reason, the `fcrepoMockEndpoint` doesn't get its two required calls. I don't see why it shouldn't do that (I guess some magic in some property or some naming?). Additionally, running the integration test against Docker works normally. So I need some guidance where to find the error.
